omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,199 @@
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests CRUD operations for {@link AutoGenerationOption}s via web service calls
+ */
+public class AutoGenerationOptionControllerTest extends MainResourceControllerTest {
+	
+	@Autowired
+	private IdentifierSourceService identifierSourceService;
+	
+	public static final String NEW_LOCATION = "9356400c-a5a2-4532-8f2b-2361b3446eb8";
+	
+	public static final String NEW_SOURCE = "0d47284f-9e9b-4a81-a88b-8bb42bc0a901";
+	
+	public static final String LOCATION = "8d6c993e-c2cc-11de-8d13-0010c6dffd0f";
+	
+	public static final String SOURCE = "0d47284f-9e9b-4a81-a88b-8bb42bc0a902";
+	
+	public static final String AUTO_GENERATION_OPTION_UUID = "2";
+	
+	public static final String IDENTIFIER_TYPE_UUID = "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c";
+	
+	public static final String UNKNOWN_LOCATION = "8d6c993e-c2cc-11de-8d13-0010c6dffd0f";
+	
+	@Before
+	public void before() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+	}
+	
+	@Override
+	public String getURI() {
+		return "idgen/autogenerationoption";
+	}
+	
+	@Override
+	public String getUuid() {
+		return AUTO_GENERATION_OPTION_UUID;
+	}
+	
+	private int getId() {
+		return Integer.parseInt(getUuid());
+	}
+	
+	@Override
+	public long getAllCount() {
+		Integer allAutoGenerationOptions = 0;
+		for (PatientIdentifierType patientIdentifierType : Context.getPatientService().getAllPatientIdentifierTypes()) {
+			allAutoGenerationOptions += identifierSourceService.getAutoGenerationOptions(patientIdentifierType).size();
+		}
+		return allAutoGenerationOptions;
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOption() throws Exception {
+		long originalCount = getAllCount();
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("location", LOCATION);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", true);
+		autoGenerationOption.add("manualEntryEnabled", true);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		assertNotNull( PropertyUtils.getProperty(newAutoGenerationOption, "source"));
+		assertNotNull( PropertyUtils.getProperty(newAutoGenerationOption, "location"));
```

Comment:
Instead of assertNotNull, wouldn't assertEquals be more precise? Because it may not be null, but have a value different from the one you expect. Not so?